process/dns.go

```diff
@@ -2,6 +2,8 @@ package process
 
 type DNSEncoder interface {
 	Encode(dns map[string]*DNSEntry) []byte
+	EncodeMapped(dns map[string]*DNSDatabaseEntry) []byte
+	EncodeDomainDatabase(names []string) []byte
```

Comment:
Can there be documentation added to this? It's not clear what any of these do, and what the return types are from the signatures.